google/gax/api_callable.py

```diff
@@ -382,6 +382,28 @@ def inner(*args, **kwargs):
     return inner
 
 
+def _merge_options_metadata(options, settings):
```

Comment:
If I recall correctly, the ordering was an optimization for server-side performance. As you noted, I don't think it makes any difference semantically. @lukesneeringer and @bjwatson have much more context here than I do.